README.md
# Forgotten-Armament

[![Latest Release](https://img.shields.io/github/v/release/gibberlings3/Forgotten-Armament?include_prereleases)](https://github.com/Gibberlings3/Forgotten-Armament/releases/latest)
![Language](https://img.shields.io/static/v1?label=language&message=english%20%7C%20french&color=informational)
![Platform](https://img.shields.io/static/v1?label=platform&message=windows%20%7C%20macos%20%7C%20linux&color=informational)

[A Gibberlings Three Mod](https://www.gibberlings3.net/)

**Author**: [morpheus562](https://www.gibberlings3.net/profile/11591-morpheus562/)

**On the Web**: [Home Page](https://www.gibberlings3.net/mods/items/forgotten_armament/) and [Discussion Topic](https://www.gibberlings3.net/forums/topic/33923-forgotten-armament-beta/)

## Introduction

Forgotten Armament aims to bring many low to mid-tier magical weapons, armors, and items into the games of Baldur's Gate: Enhanced Edition, Baldur's Gate II: Enhanced Edition, and the Enhanced Edition Trilogy (EET). The focus is to bring competitive mid-tier items, inspired by IWD and NWN2, into the game. One aspect of this mod is to ensure all weapons, armor, and items are equally represented and have steady progression paths for players throughout the series. Additionally throughout the games, many a mid-level enemy is poorly equipped with either non-magical or generically magical (+1 non-named) weapons and armor while similarly leveled players have +2/+3 weapons and gear. Forgotten Armament will provide many of these enemies with unique +1/+2/+3 weapons and gear providing for improved encounters with properly equipped foes.   

Through time, this has also branched into a more of an all-encompassing item mod for me. With this, I update all items to fall more in-line with their NWN2 versions (3.5 edition) and move away from the traditional Baldur's Gate style of setting an ability score to "x" amount and instead incrementing.

## Components

### Install Forgotten Armament Items

This installs all new items throughout the game. 

A list of the item descriptions is [here](https://github.com/Gibberlings3/Forgotten-Armament/blob/main/ITEM-DESCRIPTIONS.md).

A list of the item locations is [here](https://github.com/Gibberlings3/Forgotten-Armament/blob/main/ITEM-LOCATIONS.md).

#### OPTIONAL SUBCOMPONENT - Update Existing Items in Addition to Adding New Ones

This subcomponent is optional and updates many weapons, armors, and items within the game. This is actively in development, and the current changes to items can be found [here](https://github.com/Gibberlings3/Forgotten-Armament/blob/main/UPDATED-ITEM-DESCRIPTIONS.md).

### Replace Boots of Speed (Paws of the Cheetah)

This component replaces Boots of Speed (Paws of the Cheetah) with other boot options.

### Add Breach Scrolls to Sorcerous Sundries

This component adds a couple of Breach scrolls to Sorcerous Sundries.

### NWN2 Style Weapons

This component updates all weapons to match weapons in NWN2 to include damage, damage type, and critical threat range/damage. This includes all weapons added by mods. A detailed description of the new base weapon stats is [here](https://github.com/Gibberlings3/Forgotten-Armament/blob/main/NWN2_STYLE_WEAPONS.md).

### Update Items Setting an Ability Score to Increment the Increase

This component updates all items to increment increases to ability scores instead of setting them. For example, the Belt of Hill Giant normally sets Strength to 19. This component will update the item to increment Strength by +2. All items that set ability scores, included those added by other mods, will be updated to increment with this component.

## Compatibility

If you are upgrading from an earlier version of this mod, I recommend uninstalling all mod components and replacing all files with the new ones before doing a fresh install.

I have reviewed and deconflicted any compatibility issues with multiple item mods to include: Made in Heaven: Item Pack, Sod2BG2 items, and The Calling Exotic Item Pack. If you do notice a compatibility issue with other mods, especially the ones listed here, please let me know so I can correct it.

NWN2 Style Weapons and Update Items Setting an Ability Score to Increment the Increase components should be installed AFTER any mods that add new weapons or items to the game.

Do not install the NWN2 Style Weapons component with other mods that alter weapons like Scales of Balance.  

## Special Thanks

- Auve for support in the creation, inspiration, testing, and feedback for this mod.
- JohnBob for translating into French.
- Bubb and CamDawg_G3 for their usual expertise, guidance, and assistance with coding.


changelog.md
## Version 0.7.1 BETA

June 26, 2022

### Changed

- Corrected bug with NWN style weapons.

## Version 0.7 BETA

June 23, 2022

### Changed

- Added sub-component to update over 100 existing items.

## Version 0.6.1 BETA

June 1, 2022

### Changed

- Corrected bug with weapons receiving extra damage on critical hit.

## Version 0.6 BETA

May 31, 2022

### Changed

- Added component to update all stat increasing items to increment instead of set the value.
- Added component to update all weapons to NWN2 Style stats.
- Added component to add breach scrolls to Sorcerous Sundries.

## Version 0.5 BETA

March 27, 2022

### Changed

- Updated Spears to deal 1d8 damage instead of 1d6 damage. Additionally, on critical hits Spears will now deal and additional +1d8 damage.

## Version 0.4.1 BETA

March 27, 2022

### Changed

- Corrected bug where a non-valid armor was assigned to Revanek.

## Version 0.4 BETA

March 4, 2022

### Changed

- Added French translations courtesy of JohnBob.

## Version 0.3 BETA

February 4, 2022

### Changed

- Removed Weapon Style Changes, these are now part of my Skills and Abilities mod.
- Updated TRA files.

## Version 0.2.2 BETA

January 17, 2022

### Changed

- Corrected behavior for Single Weapon Style Bonus.

## Version 0.2 BETA

January 7, 2022

### Changed

- Added in updated weapon style proficiencies.

## Version 0.1 BETA

January 5, 2022

### Changed

- Initial release for G3.


forgotten-armament/dlg/mo2h15.d
EXTEND_BOTTOM BOTSMITH 4
IF ~PartyHasItem("sw2h15")~ THEN GOTO MO#PartyHasAtLeastOneItem END
  
APPEND BOTSMITH
IF ~~ THEN BEGIN MO#PartyHasAtLeastOneItem SAY @40010
	IF ~PartyHasItem("sw2h15")
		PartyHasItem("compon01")~ THEN GOTO MO#PartyHasAllParts
	IF ~PartyHasItem("sw2h15")
		!PartyHasItem("compon01")~ THEN GOTO MO#PartyDoesNotHaveAllParts
END 

IF ~~ THEN BEGIN MO#PartyDoesNotHaveAllParts SAY @40011
	IF ~~ THEN GOTO MO#MovingRightAlong
END 

IF ~~ THEN BEGIN MO#PartyHasAllParts SAY @40012
	IF ~PartyGoldLT(15000)~ THEN REPLY #66662 GOTO MO#MovingRightAlong
	IF ~PartyGoldGT(14999)~ THEN REPLY #66664 DO ~SetGlobal("MO#TOBItemUp","GLOBAL",2)
											 		 SetGlobal("ImpForgeStuff","GLOBAL",1)
													 TakePartyGold(15000)
													 DestroyGold(15000)
													 TakePartyItemNum("sw2h15",1)
													 DestroyItem("sw2h15")
													 TakePartyItemNum("compon01",1)
													 DestroyItem("compon01")~ GOTO 11
	IF ~~ THEN REPLY #66770 GOTO MO#MovingRightAlong
	END 

	IF ~~ THEN BEGIN MO#MovingRightAlong SAY @40013
		COPY_TRANS BOTSMITH 4
	END
END

forgotten-armament/dlg/moblk01.d
EXTEND_BOTTOM BOTSMITH 4
IF ~PartyHasItem("miscbc")~ THEN GOTO MO#PartyHasAtLeastOneItem END
  
APPEND BOTSMITH
IF ~~ THEN BEGIN MO#PartyHasAtLeastOneItem SAY @40000
	IF ~PartyHasItem("miscbc")
		PartyHasItem("compon04")~ THEN GOTO MO#PartyHasAllParts
	IF ~PartyHasItem("miscbc")
		!PartyHasItem("compon04")~ THEN GOTO MO#PartyDoesNotHaveAllParts
END 

IF ~~ THEN BEGIN MO#PartyDoesNotHaveAllParts SAY @40001
	IF ~~ THEN GOTO MO#MovingRightAlong
END 

IF ~~ THEN BEGIN MO#PartyHasAllParts SAY @40002
	IF ~PartyGoldLT(10000)~ THEN REPLY #66662 GOTO MO#MovingRightAlong
	IF ~PartyGoldGT(9999)~ THEN REPLY #66664 DO ~SetGlobal("MO#TOBItemUp","GLOBAL",1)
											 		 SetGlobal("ImpForgeStuff","GLOBAL",1)
											 		 SetGlobal("MO#TOBCraft","GLOBAL",1)
													 TakePartyGold(10000)
													 DestroyGold(10000)
													 TakePartyItemNum("miscbc",1)
													 DestroyItem("miscbc")
													 TakePartyItemNum("compon04",1)
													 DestroyItem("compon04")~ GOTO 11
	IF ~~ THEN REPLY #66770 GOTO MO#MovingRightAlong
	END 

	IF ~~ THEN BEGIN MO#MovingRightAlong SAY @40003
		COPY_TRANS BOTSMITH 4
	END
END

IF ~Global("MO#TOBCraft","GLOBAL",1)
	Global("ImpForgeStuff","GLOBAL",0)~ THEN BEGIN DO ~SetGlobal("MO#TOBCraft","GLOBAL",0)~ GOTO 89
END 
	